cogs/markov.py

```diff
@@ -162,6 +163,25 @@ def predict(self, num_words: int, guild_id: int, user_id: int = None, start_word
             return result
         return "Didn't get any results."
 
+    def add_punctuation(self, result):
+        if result == "Didn't get any results.":
+            return result
+        words = result.split()
+        if len(words) > 2:
+            idx = 0
+        else:
+            idx = random.randint(1, len(words) - 1)
+        words[idx] = words[idx] + "?\n"
+        words[len(words) - 1] = words[len(words) - 1] + "!"
+        return words
+
+    @commands.command("markov_joke")
+    @commands.cooldown(5, 30, commands.BucketType.user)
+    @commands.guild_only()
+    async def markov_joke(self, ctx, user: discord.User, words: int = 15):
```

Comment:
do we want this based off of a specific user? I mean I guess that would be pretty funny. Doesn't really help that the spooky-related stuff is in a different project than the markov-related stuff.

Reply:
> do we want this based off of a specific user?
That's a good question; I didn't put a ton of effort into understanding how the generalized chain is invoked differently than the user-specific chain.
> Doesn't really help that the spooky-related stuff is in a different project than the markov-related stuff.
Correct

Reply:
I'd update this method to end with `_user`, just like the other methods. We can also add another method that doesn't require a user to be specified:
```py
    @commands.command("markov_joke")
    @commands.cooldown(5, 30, commands.BucketType.user)
    @commands.guild_only()
    async def markov_joke(self, ctx, words: int = 20):
        """
        Replies with a predicted phrase morphed into a 'joke' from the messages of all users.
        """
        async with ctx.channel.typing():
            await ctx.send(self.add_punctuation(self.predict(words, ctx.guild.id)))
```

Reply:
See new commit with requested changes :+1: